frame/assets/src/lib.rs

```diff
@@ -561,7 +565,8 @@ pub mod pallet {
 			let origin = ensure_signed(origin)?;
 			let who = T::Lookup::lookup(who)?;
 
-			let f = DebitFlags { keep_alive: false, best_effort: true };
+			let f = DebitFlags { keep_alive: false, ignore_freezer: false };
+			let amount = amount.min(Self::reducible_balance(id, &who, f)?);
 			let burned = Self::do_burn(id, &who, amount, Some(origin), f)?;
 			Self::deposit_event(Event::Burned(id, who, burned));
```

Comment:
Emitted in `do_burn`: https://github.com/paritytech/substrate/blob/efd262f/frame/assets/src/functions.rs#L345
```suggestion
			let _ = Self::do_burn(id, &who, amount, Some(origin), f)?;
```